src/session-pool.ts

```diff
@@ -865,21 +892,35 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
    * @returns {Promise<Session>}
    */
   async _getSession(startTime: number): Promise<Session> {
+    const span = getActiveOrNoopSpan();
+    span.addEvent('Attempting to get session');
+
     if (this._hasSessionUsableFor()) {
-      return this._borrowNextAvailableSession();
+      span.addEvent('Cache hit: has usable session');
+      const sessPromise = this._borrowNextAvailableSession();
```

Comment:
why this change is required ?  to save results in const 

directly use `return this._borrowNextAvailableSession();`

Reply:
Sorry about that, it was a vestige from prior code that did something on the promise. Fixed, thanks!